### PR TITLE
Base XCTestCase

### LINKED ISSUE #3175: Simplify injecting shared mocks in tests
# Problem

In many of our `XCTestCase` subclasses we setup some mocked globals in `setUp` method. For example:
```swift
DirectionsCredentials.injectSharedToken(.mockedAccessToken)
```

These setups are required, or the tests won't run or crash.

Scattering the same code in many places is error-prone, time-consuming, cumbersome, and heavily against common architectural approaches. 

# Solution

Introduce an XCTestCase subclass which will be used in all the other test cases. 

This subclass will allow us to centralize common setup/teardown logic without repeating it in many places. 

@mapbox/navigation-ios

### PR DESCRIPTION
**Pull request by S2Ler**
Fixed #3175

### CODE REVIEW

File: Tests/MapboxNavigationTests/XCTestCase.swift [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import Foundation
 import XCTest
+import TestHelper
```
Comment (azarovalex) [inline Tests/MapboxNavigationTests/XCTestCase.swift:3]: do we really need this import?

**Review by azarovalex — APPROVED** [review]
lgtm

File: Tests/MapboxNavigationTests/XCTestCase.swift (hunk 1 above)
Comment (S2Ler) [reply Tests/MapboxNavigationTests/XCTestCase.swift:3]: Probably not anymore, thanks for catching, I will check.

Comment (Udumft) [inline Tests/MapboxNavigationTests/XCTestCase.swift:3]: Why is this import was added?

**Review by Udumft — APPROVED** [review]
LGTM with small question

File: Tests/MapboxNavigationTests/XCTestCase.swift (hunk 1 above)
Comment (S2Ler) [reply Tests/MapboxNavigationTests/XCTestCase.swift]: It used to be needed, leftover.

**Pull request merged**